src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,454 @@
+
+    def _compute_transaction_set(self, trade_history: Dict[str, Dict[str, Union[str, int, None, List[str]]]], ledger: Dict[str, Dict[str, Union[str, int, None, List[str]]]]) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        unhandled_types: Dict[str, str] = {}
+        for key in ledger:
+            record: Dict[str, str] = ledger[key]
+            self.__logger.debug("Ledger record: %s", record)
+
+            timestamp_value: str = self._rp2_timestamp_from_seconds_epoch(record[_TIMESTAMP])
+
+            is_fiat_asset: bool = record[_ASSET] in _KRAKEN_FIAT_LIST
+
+            amount: RP2Decimal = abs(RP2Decimal(record[_AMOUNT]))
+            asset_base: str = self.base_id_to_base[record[_ASSET]]
+            raw_data = str(record)
+
+            if record[_TYPE] == _WITHDRAWAL or record[_TYPE] == _DEPOSIT:
+                is_deposit: bool = record[_TYPE] == _DEPOSIT
+                is_withdrawal: bool = record[_TYPE] == _WITHDRAWAL
+                spot_price: str = Keyword.UNKNOWN.value
+
+                result.append(
+                    IntraTransaction(
+                        plugin=self.__PLUGIN_NAME,
+                        unique_id=Keyword.UNKNOWN.value,
+                        raw_data=raw_data,
+                        timestamp=timestamp_value,
+                        asset=asset_base,
+                        from_exchange=self.__EXCHANGE_NAME if is_withdrawal else Keyword.UNKNOWN.value,
+                        from_holder=self.account_holder if is_withdrawal else Keyword.UNKNOWN.value,
+                        to_exchange=self.__EXCHANGE_NAME if is_deposit else Keyword.UNKNOWN.value,
+                        to_holder=self.account_holder if is_deposit else Keyword.UNKNOWN.value,
+                        spot_price=spot_price,
+                        crypto_sent=Keyword.UNKNOWN.value if is_deposit else str(amount),
+                        crypto_received=str(amount) if is_deposit else Keyword.UNKNOWN.value,
+                        notes=key,
+                    )
+                )
+                continue
+
+            crypto_fee: str = '0' if is_fiat_asset else record[_FEE]
+            fiat_fee: Optional[str] = record[_FEE] if is_fiat_asset else None
+
+            if record[_TYPE] == _TRADE and not is_fiat_asset:
+                self.__logger.debug("Trade history record: %s", trade_history[record[_REFID]])
+
+                asset_quote: str = self._client.markets_by_id[trade_history[record[_REFID]][_PAIR]][_QUOTE]
+                is_quote_asset_fiat: bool = asset_quote in _KRAKEN_FIAT_LIST
+
+                spot_price = trade_history[record[_REFID]][_PRICE] if is_quote_asset_fiat else Keyword.UNKNOWN.value
+                transaction_type: str = Keyword.BUY.value if RP2Decimal(record[_AMOUNT]) > ZERO else Keyword.SELL.value
+
+                unique_id: str = Keyword.UNKNOWN.value if spot_price is Keyword.UNKNOWN.value else key
```

Comment:
This logic still makes the value of `unique_id` depend on the value of `spot_price`, which seems suspicious. Can `unique_id` be initialized regardless of what `spot_price` is, or is there a Kraken-specific reason why this is not possible?

Reply:
I'll have to investigate. This is from the swirl when I was getting the wrong prices set to the wrong dates. In other comments, I notate that Kraken uses 'XBT' for 'BTC' and other assets in similar boat which confuses the pair converter. I'm going through the testing phase of the fix right now (its an ordeal).

Reply:
@eprbell you were right! It was my fault. When the erroneous spot_prices were happening I was searching for a solution and that was added by mistake.